packages/fiber/src/core/loop.ts

```diff
@@ -3,17 +3,18 @@ import { Root } from './renderer'
 import { RootState, Subscription } from './store'
 
 type GlobalRenderCallback = (timeStamp: number) => void
+type SubItem = {callback: GlobalRenderCallback}
 
-function createSubs(callback: GlobalRenderCallback, subs: GlobalRenderCallback[]): () => void {
-  const index = subs.length
-  subs.push(callback)
-  return () => void subs.splice(index, 1)
+function createSubs(callback: GlobalRenderCallback, subs: Set<SubItem>): () => void {
+  const sub = {callback}
```

Comment:
looks good i think we should use set, but just that, why do you wrap it into an object here?

Reply:
If you don't use object to wrap the callback, pass in the identical callback function more than once and will only one is left. 
for example just like the below.
```js
const log = (ts) = console.log('log', ts);
addEffect(log);
addEffect(log);
addEffect(log);
```
If so, it will confuse the developers.

such sutuation may be rare, please give me a message if you don't think it's necessary.

sincerely.